From 6ac7ad8a2ce59bebcb5b5085fa48c354afff539f Mon Sep 17 00:00:00 2001
From: Vincent Penquerc'h <vincent.penquerch@collabora.co.uk>
Date: Tue, 16 Aug 2011 12:16:22 +0000
Subject: flacparse: fix off by one in frame size check

Yes, I was tracking another bug and the small test file I generated
to test with improbably just happened to trigger this, with a second
and last frame of 1615 bytes.

https://bugzilla.gnome.org/show_bug.cgi?id=656649
---
(limited to 'gst/audioparsers/gstflacparse.c')

diff --git a/gst/audioparsers/gstflacparse.c b/gst/audioparsers/gstflacparse.c
index 0c6c529..3011bf0 100644
--- a/gst/audioparsers/gstflacparse.c
+++ b/gst/audioparsers/gstflacparse.c
@@ -581,7 +581,7 @@ gst_flac_parse_frame_is_valid (GstFlacParse * flacparse,
   data = GST_BUFFER_DATA (buffer);
   size = GST_BUFFER_SIZE (buffer);
 
-  if (size <= flacparse->min_framesize)
+  if (size < flacparse->min_framesize)
     goto need_more;
 
   header_ret =
--
cgit v0.9.0.2-2-gbebe

